src/lib/common/globals.cpp

```diff
@@ -606,6 +606,25 @@ double parse8601Time(const std::string& ss)
   time.tm_min  = m;        // 0-59
   time.tm_sec  = (int) s;  // 0-61 (0-60 in C++11)
 
+  const int minYear = 0;
+  const int minMonth = 0;
+  const int maxMonth = 11;
+  const int minDay = 1;
+  const int maxDay = 31;
+  const int minHour = 0;
+  const int maxHour = 23;
+  const int minMinute = 0;
+  const int maxMinute = 59;
+  const int minSecond = 0;
+  const int maxSecond = 59;
+
+  if (time.tm_year < minYear || time.tm_mon < minMonth || time.tm_mon > maxMonth || time.tm_mday < minDay ||
+      time.tm_mday > maxDay || time.tm_hour < minHour || time.tm_hour > maxHour || time.tm_min < minMinute ||
+      time.tm_min > maxMinute || time.tm_sec < minSecond || time.tm_sec > maxSecond)
```

Comment:
What about 2024-02-30? As far as I understand, this will pass this check, but February 30th is not a valid date ;)

Maybe you should make a function `maxDay(month, year)` which return the maxDay based on month and year. Basically:

* January: 31
* February: 29 in case of leap-year, 28 otherwise. Maybe you need an additional `isLeapYear(year)` function for this (maybe this reference is useful to implement it https://www.geeksforgeeks.org/program-check-given-year-leap-year)
* March: 31
* April: 30
* May: 31
* June: 30
* July: 31
* August: 31
* September: 30
* October: 31
* November: 30
* December: 31

Please include 2024-02-29 (should be correctly accepted) and 2023-02-29 (should return error) in your tests to check it works as expected.

Reply:
Fixed in [1a824bf](https://github.com/telefonicaid/fiware-orion/pull/4554/commits/1a824bf514b1b75fb43b7320ba893211bfc4666c)